### PR TITLE
#380: Update CMake config.

### LINKED ISSUE #380: About setting features after Crow is installed
@autoantwort wrote [this](https://github.com/CrowCpp/Crow/commit/258e0f50df76717e5a2e90f14c69522316b040e0#commitcomment-69885951). (Below is a copy)

>  Hm I don't know if that is a good idea. If vcpkg installs the package with openssl and zlib enabled the consuming user still have to set `CROW_ENABLE_COMPRESSION` and `CROW_ENABLE_SSL` before the `find_package` call?

@luca-schlecker what's your opinion on this?

### PR DESCRIPTION
**Pull request by luca-schlecker**
This is the follow-up of and partially fixes #380.

Crow's features (ssl and compression) can now be enabled through `CROW_FEATURES` instead of their own CMake variables.
The default features are the ones Crow was installed with but can be overridden by setting `CROW_FEATURES` before the `find_package` call.

```cmake
cmake_minimum_required(VERSION 3.17)
project(CrowTest)

add_executable(test main.cpp)

# Some examples:
# set(CROW_FEATURES "")
# set(CROW_FEATURES "ssl;compression")
# set(CROW_FEATURES ssl compression)
# set(CROW_FEATURES ssl)
find_package(Crow REQUIRED)
target_link_libraries(test Crow::Crow)
```

Using the CMake CLI features are given as a semicolon-separated list:
```bash
cmake .. -DCROW_FEATURES="ssl;compression"
```

Documentation needs an update to align with the changes made in this PR.

Signed-off-by: Luca Schlecker <luca.schlecker@hotmail.com>

### CODE REVIEW

**Comment by luca-schlecker** [comment]
Yes, the definitions inside the code stayed the same, it's just the CMake variable that changed, and the default features are now the ones Crow was installed with. For example, `crow_all.h` is not influenced by this change.

File: CMakeLists.txt [hunk 1]
```diff
@@ -92,11 +92,11 @@ endif()
...
 if(NOT MSVC AND CROW_BUILD_TESTS)
-	if(NOT CROW_ENABLE_COMPRESSION)
-		message(STATUS "Compression tests are omitted. (Configure with CROW_ENABLE_COMPRESSION=ON to enable them)")
+	if(NOT "compression" IN_LIST CROW_FEATURES)
+		message(STATUS "Compression tests are omitted. (Configure with CROW_FEATURES containing 'compression' to enable them)")
 	endif()
-	if(NOT CROW_ENABLE_SSL)
-		message(STATUS "SSL tests are omitted. (Configure with CROW_ENABLE_SSL=ON to enable them)")
+	if(NOT "ssl" IN_LIST CROW_FEATURES)
+		message(STATUS "SSL tests are omitted. (Configure with CROW_FEATURES containen 'ssl' to enable them)")
```
Comment (The-EDev) [inline CMakeLists.txt:99]:
```suggestion
		message(STATUS "SSL tests are omitted. (Configure with CROW_FEATURES containing 'ssl' to enable them)")
```

**Review by The-EDev — APPROVED** [review]
Everything seems good, I just wanted to know how do I set the default values during installation (So I can update the packages)?

**Comment by The-EDev** [comment]
Hmmm, I understand.

**Comment by The-EDev** [comment]
Sorry about all the branch merges by the way

**Review by The-EDev — APPROVED** [review]

**Comment by luca-schlecker** [comment]
All good, but that's the reason I rebase and force push instead.

**Pull request merged**